common/pooll2txapi.go

```diff
@@ -0,0 +1,147 @@
+package common
+
+import (
+	"encoding/json"
+	"math/big"
+	"time"
+
+	ethCommon "github.com/ethereum/go-ethereum/common"
+	"github.com/iden3/go-iden3-crypto/babyjub"
+)
+
+// PoolL2TxAPI represents a L2 Tx pool with extra metadata used by the API
+type PoolL2TxAPI struct {
+	ItemID               uint64
+	TxID                 TxID
+	Type                 TxType
+	FromIdx              Idx
+	EffectiveFromEthAddr *ethCommon.Address
+	EffectiveFromBJJ     *babyjub.PublicKeyComp
+	ToIdx                *Idx
+	EffectiveToEthAddr   *ethCommon.Address
+	EffectiveToBJJ       *babyjub.PublicKeyComp
+	Amount               big.Int
+	Fee                  FeeSelector
+	Nonce                Nonce
+	State                PoolL2TxState
+	MaxNumBatch          uint32
+	Info                 *string
+	ErrorCode            *int
+	ErrorType            *string
+	Signature            babyjub.SignatureComp
+	Timestamp            time.Time
+	RqFromIdx            *Idx
+	RqToIdx              *Idx
+	RqToEthAddr          *ethCommon.Address
+	RqToBJJ              *babyjub.PublicKeyComp
+	RqTokenID            *TokenID
+	RqAmount             big.Int
+	RqFee                *FeeSelector
+	RqNonce              *Nonce
+	Token                struct {
+		TokenID          TokenID
+		TokenItemID      uint64
+		TokenEthBlockNum int64
+		TokenEthAddr     ethCommon.Address
+		TokenName        string
+		TokenSymbol      string
+		TokenDecimals    uint64
+		TokenUSD         *float64
+		TokenUSDUpdate   *time.Time
+	}
+}
+
+// MarshalJSON is used to convert PoolL2TxAPI in JSON
+func (tx PoolL2TxAPI) MarshalJSON() ([]byte, error) {
+	type jsonToken struct {
+		TokenID          TokenID           `json:"id"`
+		TokenItemID      uint64            `json:"itemId"`
+		TokenEthBlockNum int64             `json:"ethereumBlockNum"`
+		TokenEthAddr     ethCommon.Address `json:"ethereumAddress"`
+		TokenName        string            `json:"name"`
+		TokenSymbol      string            `json:"symbol"`
+		TokenDecimals    uint64            `json:"decimals"`
+		TokenUSD         *float64          `json:"USD"`
+		TokenUSDUpdate   *time.Time        `json:"fiatUpdate"`
+	}
+	type jsonFormat struct {
+		ItemID               uint64                 `json:"itemId"`
+		TxID                 TxID                   `json:"id"`
+		Type                 TxType                 `json:"type"`
+		FromIdx              Idx                    `json:"fromAccountIndex"`
+		EffectiveFromEthAddr *ethCommon.Address     `json:"fromHezEthereumAddress"`
+		EffectiveFromBJJ     *babyjub.PublicKeyComp `json:"fromBJJ"`
+		ToIdx                *Idx                   `json:"toAccountIndex"`
+		EffectiveToEthAddr   *ethCommon.Address     `json:"toHezEthereumAddress"`
+		EffectiveToBJJ       *babyjub.PublicKeyComp `json:"toBJJ"`
+		Amount               big.Int                `json:"amount"`
```

Comment:
This fields are not following the API format

Reply:
Hello. Sorry, i did not understood this comment and others 👇 . As you can see here https://github.com/hermeznetwork/hermez-node/pull/1049/commits/20f8497c0118523c4f77a1d32d4c9fe8bd070ee7#diff-d529406a2271fd28bec09e70ede2455ba1b484a87d25250e37f3f6390dfc143dL60 I just copied the JSON field name as is, no changes.